include/ed/helpers/msg_conversions.h

```diff
@@ -0,0 +1,134 @@
+#ifndef ED_HELPERS_MSG_CONVERSIONS_H_
+#define ED_HELPERS_MSG_CONVERSIONS_H_
+
+#include "ed_msgs/EntityInfo.h"
+#include "ed/entity.h"
+
+#include "geolib/Shape.h"
+#include "geolib/Box.h"
+#include "geolib/CompositeShape.h"
+#include "geolib/datatypes.h"
+#include "geolib/ros/msg_conversions.h"
+
+#include "tue/config/yaml_emitter.h"
+
+#include <shape_msgs/SolidPrimitive.h>
+
+namespace ed {
+
+// ------------------------------ TO ROS ------------------------------
+
+/**
+ * @brief converting geo::ShapeConstPtr to ed_msgs::SubVolume message
+ * @param shape geo::ShapeConstPtr as input
+ * @param msg filled ed_msgs::SubVolume message as output
+ */
+void convert(const geo::ShapeConstPtr shape, ed_msgs::SubVolume& sub_Volume)
+{
+    geo::Vector3 min = shape->getBoundingBox().getMin();
+    geo::Vector3 max = shape->getBoundingBox().getMax();
+
+    geo::Vector3 pos = (min + max)/2;
+    geo::Vector3 size = max - min;
+
+    geo::convert(pos, sub_Volume.center_point.point);
+
+    shape_msgs::SolidPrimitive solid;
+    sub_Volume.geometry.type = sub_Volume.geometry.BOX;
+    sub_Volume.geometry.dimensions.resize(3, 0);
+    sub_Volume.geometry.dimensions[solid.BOX_X] = size.x;
+    sub_Volume.geometry.dimensions[solid.BOX_Y] = size.y;
+    sub_Volume.geometry.dimensions[solid.BOX_Z] = size.z;
+}
+
+/**
+ * @brief converting ed::Entity to ed_msgs::EntityInfo message
+ * @param e ed::Entity as input
+ * @param msg filled ed_msgs::EntityInfo message as output
+ */
+void convert(const ed::Entity& e, ed_msgs::EntityInfo& msg) {
+    msg.id = e.id().str();
+    msg.type = e.type();
+
+    for(std::set<std::string>::const_iterator it = e.types().begin(); it != e.types().end(); ++it)
+        msg.types.push_back(*it);
+
+    msg.existence_probability = e.existenceProbability();
+
+    // Convex hull
+    const ed::ConvexHull& convex_hull = e.convexHull();
+    if (!convex_hull.points.empty())
+    {
+        msg.z_min = convex_hull.z_min;
+        msg.z_max = convex_hull.z_max;
+
+        msg.convex_hull.resize(convex_hull.points.size());
+        for(unsigned int i = 0; i < msg.convex_hull.size(); ++i)
+        {
+            msg.convex_hull[i].x = convex_hull.points[i].x;
+            msg.convex_hull[i].y = convex_hull.points[i].y;
+            msg.convex_hull[i].z = 0;
+        }
+    }
```

Comment:
You should initialize the `msg` fields also when the hull has no points I think, or you cannot reliably read the message to see there is no hull.

Reply:
This is done by the constructor of the msg.

Reply:
but you're not constructing a `msg` here, so there is no way of knowing it has the proper values.

Reply:
This is the output of the service call to `simple_query`:
```
entities: 
  - 
    id: "exit_1_rips"
    type: "waypoint"
    types: [waypoint]
    existence_probability: 1.0
    last_update_time: 
      secs: 0
      nsecs:         0
    creation_time: 
      secs: 0
      nsecs:         0
    has_shape: False
    has_pose: True
    pose: 
      position: 
        x: -1.48
        y: -0.031
        z: 0.0
      orientation: 
        x: 0.0
        y: 0.0
        z: 0.999999998927
        w: 4.632679488e-05
    data: ''
    flags: []
    convex_hull: []
    z_min: 0.0
    z_max: 0.0
    volumes: []
```
`convert` is called with a reference of the msg. So the message is constructed somewhere else.
The constructor of the message is:
```cpp
EntityInfo_()
    : id()
    , type()
    , types()
    , existence_probability(0.0)
    , last_update_time()
    , creation_time()
    , has_shape(false)
    , has_pose(false)
    , pose()
    , data()
    , flags()
    , convex_hull()
    , z_min(0.0)
    , z_max(0.0)
    , volumes()  {
    }
```
So I still think the message is initialized correct. So the convex_hull will be an empty vector.